### PR TITLE
Cloud 708 ig benchmark

### PR DESCRIPTION
**Pull request by lee-baines**
Updated Gatling docker container simulations with IG benchmark scenario.
Added IG benchmark sample configuration to sample folder.

### CODE REVIEW

File: docker/gatling/simulations/ig/IGReverseProxyWebSim.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    val userPoolSize: Integer = Integer.getInteger("users", 3)
+    val concurrency: Integer = Integer.getInteger("concurrency", 200)
+    val duration: Integer = Integer.getInteger("duration", 60)
+    val warmup: Integer = Integer.getInteger("warmup", 3)
+    val igHost: String = System.getProperty("ig_host", "openig.example.forgeops.com")
+    val igPort: String = System.getProperty("ig_port", "80")
+    val igProtocol: String = System.getProperty("ig_protocol", "http")
+
+    val igUrl: String = "http://" + igHost + ":" + igPort
+    val random = new util.Random
```
Comment (Atomic182) [inline docker/gatling/simulations/ig/IGReverseProxyWebSim.scala]: Do you need this random ?

**Comment by gandru** [comment]
shouldn't it be possible to store scala files outside the image and run gatling pod with scala file as a parameter ? 
(easier/quickest for debug/write new test)

File: docker/gatling/simulations/ig/IGReverseProxyWebSim.scala (hunk 1 above)
Comment (lee-baines) [reply docker/gatling/simulations/ig/IGReverseProxyWebSim.scala]: o, I'll remove that

**Comment by lee-baines** [comment]
Good point about the scala file being a parameter.  Will discuss that and make a separate story if necessary.

**Comment by gandru** [comment]
> Good point about the scala file being a parameter. Will discuss that and make a separate story if necessary.

cool thanks, can you put me in the loop ? I have few comments/feedbacks to add as well

**Comment by wstrange** [comment]
Possible approaches to externalizing the scala test files:
- Add them as a configmap in helm, and mount in the pod as a volume (see the openidm, etc, charts)
- Have them stored in git (forgeops-init/benchmark/ig-test/*.scala).   Git clone them in an init container before the benchmark starts

**Comment by lee-baines** [comment]
> Possible approaches to externalizing the scala test files:
> 
> * Add them as a configmap in helm, and mount in the pod as a volume (see the openidm, etc, charts)
> * Have them stored in git (forgeops-init/benchmark/ig-test/*.scala).   Git clone them in an init container before the benchmark starts

I think option 1 is better. Saves adding a new folder in forgeops-init(just got it down to 2).  
I've created a story for this https://bugster.forgerock.org/jira/browse/CLOUD-775. Guillaume - I've added you as watcher so you can follow progress.  I'll do it this sprint.

**Pull request merged**